x-pack/qa/evil-tests/build.gradle

```diff
@@ -6,6 +6,7 @@ dependencies {
 }
 
 unitTest {
+    systemProperty 'java.locale.providers','SPI,COMPAT'
```

Comment:
what would happen with jdk8 ? COMPAT is not known there. as per http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/sun/util/locale/provider/LocaleProviderAdapter.java#l135 it should throw some exceptions (although when I tried to reproduce it it did not fail)
can we make this conditional?
Also, do we need SPI there? SPI options is not used in 6.8 by ES as we do not provide IsoCalendarProvider (because SPIs have to be shipped with jre)

Reply:
Sorry I forgot that we talked about this. I'll make it conditional for java 9 and later.

As for `SPI`, you are right that it is not needed here, I am adding it here with the attempt to be consistent with:
https://github.com/elastic/elasticsearch/blob/029903173e981aadbfd615cdacfaa43867b6b92f/gradle/ide.gradle#L74-L76